system/lib/wasmfs/paths.cpp

```diff
@@ -85,26 +85,29 @@ ParsedPath getParsedPath(std::vector<std::string> pathParts,
 
     // Find the relevant child for this path part.
     std::shared_ptr<File> child;
+    std::shared_ptr<File> parent;
     if (*pathPart == ".") {
       child = lockedCurr.unlocked();
+      parent = lockedCurr.getParent();
     } else if (*pathPart == "..") {
       if (curr != wasmFS.getRootDirectory()) {
         child = lockedCurr.getParent()->cast<Directory>();
       } else {
         child = lockedCurr.unlocked();
```

Comment:
Could this just be `child = curr`? Or even `child = std::move(curr)`? Also, it might be good to reverse the condition here.

Reply:
Good ideas, done.